doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,352 @@
+  - Interoperate with non-RIOT nodes or gateways.
+  - Collect data on timescales of the order of hours or longer.
+  - Potentially be able to send data over long ranges.
+  - Potentially communicate with a smartphone.
+  - Potentially operate for years without power infrastructure or maintenance.
+
+## 2.2. Rapid prototyping, research, and experimentation
+
+In experimentation and hacking situations, development needs to be easily
+accessible, and allow a short development time and quick results. This means
+that the software and hardware should:
+
+  - Let users prototype easily.
+  - Let users easily load and run simple applications.
+  - Let users easily set up networking.
+  - Let users easily port third party libraries.
+  - Be highly customizable and updateable.
+  - Support a range of plug-in sensors and actuators.
+  - Be usable with or without different features, including networking.
+  - Use commonly available, easy to use development boards.
+  - Come with an easy-to-use toolkit that has a minimum of setup time.
+  - Let users easily port programs to different devices.
+
+## 2.3. Logistic tracking
+
+Sensors that record environmental conditions can be used to manage goods in
+transit. These sensors need to:
+
+  - Sense position and condition of goods.
+  - Last for several months without charging.
+  - Send data over long ranges to regional infrastructure.
+
+## 2.4. Physical system monitoring and control
+
+Distributed networks of sensors and actuators can be employed in certain
+control applications, such as automotive systems or Industry 4.0. The nodes
+need to:
+
+  - Sense various specific properties such as pressure, torque, velocity, or
+    flow.
+  - Control various specific actuators such as motors, solenoids, or valves.
+  - Collect and send data with a low latency, or at least a well synchronized
+    timestamp.
+  - Potentially have a very low unit cost, particularly when the networks are
+    implemented in mass market products such as automobiles.
+  - Potentially run control algorithms themselves.
+
+## 2.5. Edge systems for building management and automation
+
+Various sensing and environmental control tasks can be done by edge nodes in
+buildings. The nodes need to be able to:
+
+  - Sense temperature, light, humidity, pressure, current, etc.
+  - Control temperature, light, ventilation, etc.
+  - Connect to the building management system, via wired or wireless connectors.
+  - Deliver data with controllable timing and accuracy.
+
+## 2.6. Smart home devices
+
+Easy to use devices can let consumers monitor and control their home
+automatically. The devices need to be able to:
+
+  - Sense temperature, light, humidity, smoke, etc.
+  - Integrate with home appliances to control temperature, light, power usage,
+    etc.
+  - Be usable but secure for people with no technical knowledge.
+  - Connect to a commercial home gateway.
+  - Communicate over common home IoT protocols such as ZigBee, Bluetooth, Wi-Fi,
```

Comment:
With this sentence from section 3 in mind

> RIOT should support standards once they have reached a certain level of
maturity and popularity.

I guess it is about high time we write a ZigBee stack then ;-). Question is are we able / allowed to do that (I remember there was some paywall thingy)?

Reply:
I actually was asked about this at 35C3 by @Drakulix. His use-case is to have a non-intrusive but open integration with his TRÅDFRI home automation setup.

Reply:
Thanks for the catch. AFAIU you can implement Zigbee but have to pay to get it certified (and call it ZigBee). In any case it seems that Zigbee isn't in line with our open philosophies. So, I could remove the ZigBee here? and make sure that it's clear throughout the doc that we support only *open* networking protocols?

Reply:
Zigbee removed after discussion offline